### ISSUE TITLE
Environment building fails on Windows under non-ASCII username

### ISSUE THREAD

**Issue by vassilevsky** [enkessler/childprocess#81]
The reason I opened #80 is that ChildProcess crashes when I run it. (And not some "clever" marketing BS that issue totally sounded like.)

I am Russian. When we Russians install Windows, it suggests that we use a Russian username. It even gives us an example: `Андрей`. Naturally, we all enter our names into the field and proceed, satisfied. (I actually entered `Андрей` once, when installing Windows 7 for my wife. She didn't get the joke.)

Thus, we end up having all our files in `C:\Users\Андрей` and a lot of environment variables having `Андрей` in their values because of that.

That's when things start to get weird.

Apparently, there's a bug in MRI. Not sure if it's Windows only. All values in `ENV` are in UTF-8 (which is awesome). But at the same time, their encoding is set to IBM866. And they're frozen.

Long story short, this line crashes with the IncompatibleEncodings exception:

https://github.com/jarib/childprocess/blob/master/lib/childprocess/windows/process_builder.rb#L63

I'll try to fix this situation and attach dem commits when I have some time & sleep.

P. S. I discovered this by attempting to run Vagrant.

**Comment by DavidS** [comment]
See https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L236 for one way to work around this.

**Comment by sds** [comment]
This is an old issue, but it seems like this would have likely been fixed by #134. Thanks for reporting!
